### ISSUE TITLE
Inconsistent behavior when saving `TaylorN`s with JLD2 v0.4

### ISSUE THREAD

**Issue by LuEdRaMo** [JuliaDiff/TaylorSeries.jl#373]
When saving a long `Vector{TaylorN{Float64}}` with JLD2 v0.4 (required minor version in `Project.toml`) I get inconsistent results:
```julia
v = set_variables(Float64, "dx"; order = 5, numvars = 6)
x = [exp(sum(rand(6) .* v)) for _ in 1:10_000]
begin
    for _ in 1:10
        jldsave("test.jld2"; x)
        y = JLD2.load("test.jld2", "x")
        println(count(x .!= y))
    end
end
# 3451, 1967, 965, 0, 0, 0, 2051, 0, 1663, 1305
```
But with JLD2 v0.5 (latest minor version), everything works:
```julia
# 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
```
¿@lbenet can you reproduce this?

**Comment by lbenet** [comment]
What version of TaylorSeries are you using? 

I'm getting the same inconsistencies when I use TaylorSeries v0.18.1 and JLD2 v0.4..53. Yet, if I install JLD2 v0.5 it downgrades TaylorSeries to v0.17.3, and then everything seems consistent. 

So, there seems to be an issue with the version and Project.toml in TaylorSeries v0.18.1...

**Comment by lbenet** [comment]
It seems to be an issue in TaylorSeries v0.18.1: with JLD2 v0.4..53 and TaylorSeries to v0.17.3 everything seems ok. Can you confirm this?

**Comment by lbenet** [comment]
As we discussed offline, the problem is solved by upgrading to JLD v0.5; #374 should fix it.
